changes/2020-07-13_caching-cmm-max-plaintext-length/change.md

```diff
@@ -0,0 +1,159 @@
+and if the caching CMM returns the cached materials,
+then the user may use the materials to encrypt _more_ than `N` bytes.
+The caching CMM violates its prior intent, which is unsafe.
+
+It is therefore important that the caching CMM
+carefully selects the `max plaintext length` parameter to pass to its underlying CMM.
+This change mandates that the caching CMM passes its configured `byte limit` value
+as the `max plaintext length` parameter to the underlying CMM,
+whereas previously, the caching CMM could choose any value.
+With this change in place,
+even if the underlying CMM produces materials using plaintext-length-dependent logic,
+the caching CMM receives and caches materials
+that are safe to use for up to `byte limit` bytes of plaintext.
+It follows that if a later Get Encryption Materials operation's `max plaintext length`
+does not exceed the remaining number of bytes for which the cached materials can be safely used
+(i.e.,
+the difference between the caching CMM's `byte limit`
+and the sum of `max plaintext length` values from past uses of the cached materials),
+then it is also safe for the caching CMM to return the cached materials.
+
+## Drawbacks
+
+Bypassing the cache in the caching CMM on Get Encryption Materials operations
+that do not specify `max plaintext length`
+could be confusing to the user.
+Warning the user when this happens could help prevent users from running into
+unexpected (and potentially expensive) calls to the underlying CMM,
+but we leave this as an implementation choice.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change will break any customer use case
+that depends on the Caching CMM passing no `max plaintext length` parameter
+to the underlying CMM on Get Encryption Materials calls.
+
+## Guide-level Explanation
+
+When making a Get Encryption Materials call to the caching CMM,
+the user may optionally specify a value for the `max plaintext length` parameter,
+which indicates the maximum number of bytes the user intends to encrypt
+with the materials received from the call.
+If the user does not specify a `max plaintext length` value,
+then the caching CMM will neither use cached materials
+nor cache materials from its underlying CMM.
+This helps to ensure that the number of bytes encrypted with the materials
+does not exceed the `byte limit` that the user configures on the caching CMM,
+while also maximizing the benefits of caching.
+So whenever the user can determine an appropriate upper bound for bytes they will encrypt,
+they should specify the `max plaintext length` value.
+
+If the user customizes the underlying CMM of a caching CMM,
+then whenever the caching CMM makes a Get Encryption Materials call
+to the customized underlying CMM,
+the Encryption Materials Request will have a `max plaintext length` value
+equal to the `byte limit` value configured on the caching CMM.
```

Comment:
What does "customizes the underlying CMM" mean in this case? This statement is true regardless of what underlying CMM you use.